src/SDK/MonitoringApplianceEmulator/ViewModels/ChartPropertyControlViewModel.cs

```diff
@@ -0,0 +1,130 @@
+                }
+
+                double value;
+                try
+                {
+                    value = Convert.ToDouble(dataPoint.Y, CultureInfo.InvariantCulture);
+                }
+                catch (Exception e) when (e is FormatException || e is InvalidCastException || e is OverflowException)
+                {
+                    throw new InvalidCastException($"The data point's X value '{dataPoint.X}' is not of a numeric type", e);
+                }
+
+                chartValues.Add(new DateTimeDataPoint(dateTime, value));
+            }
+
+            /*
+             * Relevant for Column Series:
+             * Since we are using DateTime.Ticks as X, the width of the bar is 1 tick and 1 tick is 1 millisecond.
+             * In order to make our bars visible we need to change the unit of the chart. For the initial view are going to use hours.
+             * There is a future task (#1380564) to create this scale dynamically according to the X values range size.
+             */
+            CartesianMapper<DateTimeDataPoint> pointMapperConfig = Mappers.Xy<DateTimeDataPoint>()
+                .X(dateTimeDataPoint => (double)dateTimeDataPoint.DateTime.Ticks / TimeSpan.FromHours(1).Ticks)
+                .Y(dateTimeDataPoint => dateTimeDataPoint.Value);
+
+            this.SeriesCollection = new SeriesCollection(pointMapperConfig);
+
+            if (chartAlertProperty.ChartType == ChartType.LineChart)
+            {
+                this.SeriesCollection.Add(new LineSeries
+                {
+                    Values = chartValues,
+                    Stroke = SeriesColor,
+                    Fill = Brushes.Transparent
+                });
+            }
+            else
+            {
+                this.SeriesCollection.Add(new ColumnSeries
+                {
+                    Values = chartValues,
+                    Stroke = SeriesColor,
+                    Fill = SeriesColor
+                });
+            }
+
+            this.XAxisFormatter = value => new DateTime((long)(value * TimeSpan.FromHours(1).Ticks)).ToString(CultureInfo.InvariantCulture);
+        }
+
+        /// <summary>
+        /// Gets the series collection.
+        /// </summary>
+        public SeriesCollection SeriesCollection
+        {
+            get
+            {
+                return this.seriesCollection;
+            }
+
+            private set
+            {
+                this.seriesCollection = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Gets the series collection.
+        /// </summary>
+        public Func<double, string> XAxisFormatter
+        {
+            get
+            {
+                return this.xAxisFormatter;
+            }
+
+            private set
+            {
+                this.xAxisFormatter = value;
+                this.OnPropertyChanged();
+            }
```

Comment:
if this is a read-only property you don't need the whole implementation
just make it an auto property

Reply:
definitely :)

---
In reply to: [231016453](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/17#discussion_r231016453) [](ancestors = 231016453)